### PR TITLE
Contracts/feature/owl outsourcing

### PR DESCRIPTION
**Pull request by josojo**
- implements proxy contract
- implements OWL airdrop from owl-token repository

-all tests are passing

### CODE REVIEW

File: test/testFunctions.js [hunk 1]
```diff
@@ -95,6 +95,8 @@ const setupTest = async (
     ethUSDPrice = 1100.0.toWei(),
   }) => {
   // Await ALL Promises for each account setup
+  await
```
Comment (Velenir) [inline test/testFunctions.js]: a wild `await` appeared

Comment (josojo) [reply test/testFunctions.js]: thx

Comment (josojo) [reply test/testFunctions.js]: fixxed thanks

File: migrations/3_deploy_internalTestsContract.js [hunk 2]
```diff
@@ -1,21 +1,23 @@
...
+    .then((p) => {
       const initParams = Promise.all([
-        dx.TUL.call(),
-        dx.OWL.call(),
-        dx.auctioneer.call(),
-        dx.ETH.call(),
-        dx.ETHUSDOracle.call(),
-        dx.thresholdNewTokenPair.call(),
-        dx.thresholdNewAuction.call(),
+        DutchExchange.at(p.address).TUL.call(),
```
Comment (Velenir) [inline migrations/3_deploy_internalTestsContract.js]: Let's do
```
const dx = DutchExchange.at(p.address)

const initParams = Promise.all([
   dx.TUL.call(),
   ...
```

Comment (josojo) [reply migrations/3_deploy_internalTestsContract.js]: sure...

I feel like a lot of quick changes have been done, just to get it work. Feel free to make any commits to get it cleaner

**Review by W3stside — APPROVED** [review]
No issues for me on anything JS related

**Comment by Velenir** [comment]
yes issues for me, no gasCost is counted for new `DutchExchange` contract
don't merge yet

**Comment by Velenir** [comment]
All good now

**Review by Velenir — APPROVED** [review]

**Review by hacker-DOM — APPROVED** [review]
looks dope

**Pull request merged**